app/containers/EmojiPicker/EmojiCategory.tsx

```diff
@@ -26,48 +26,39 @@ const renderEmoji = (emoji: IEmoji, size: number, baseUrl: string) => {
 	);
 };
 
-class EmojiCategory extends React.Component<IEmojiCategory> {
-	renderItem(emoji: IEmoji) {
-		const { baseUrl, onEmojiSelected } = this.props;
-		return (
-			<TouchableOpacity
-				activeOpacity={0.7}
-				// @ts-ignore
-				key={emoji && emoji.isCustom ? emoji.content : emoji}
-				onPress={() => onEmojiSelected(emoji)}
-				testID={`reaction-picker-${emoji && emoji.isCustom ? emoji.content : emoji}`}>
-				{renderEmoji(emoji, EMOJI_SIZE, baseUrl)}
-			</TouchableOpacity>
-		);
-	}
-
-	render() {
-		const { emojis, width } = this.props;
-
-		if (!width) {
-			return null;
-		}
-
-		const numColumns = Math.trunc(width / EMOJI_SIZE);
-		const marginHorizontal = (width - numColumns * EMOJI_SIZE) / 2;
+const EmojiCategory = React.memo(({ baseUrl, onEmojiSelected, emojis, width, tabsCount, ...props }: IEmojiCategory) => {
+	const emojiSize = width ? Math.min(width / tabsCount, MAX_EMOJI_SIZE) : MAX_EMOJI_SIZE;
+	const numColumns = Math.trunc(width ? width / emojiSize : tabsCount);
+	const renderItem = (emoji: IEmoji) => (
+		<TouchableOpacity
+			activeOpacity={0.7}
+			// @ts-ignore
+			key={emoji && emoji.isCustom ? emoji.content : emoji}
+			onPress={() => onEmojiSelected(emoji)}
+			testID={`reaction-picker-${emoji && emoji.isCustom ? emoji.content : emoji}`}>
+			{renderEmoji(emoji, emojiSize, baseUrl)}
+		</TouchableOpacity>
+	);
 
-		return (
-			<FlatList
-				contentContainerStyle={{ marginHorizontal }}
-				// rerender FlatList in case of width changes
-				key={`emoji-category-${width}`}
-				// @ts-ignore
-				keyExtractor={item => (item && item.isCustom && item.content) || item}
-				data={emojis}
-				extraData={this.props}
-				renderItem={({ item }) => this.renderItem(item)}
-				numColumns={numColumns}
-				initialNumToRender={45}
-				removeClippedSubviews
-				{...scrollPersistTaps}
-			/>
-		);
+	if (!width) {
+		return null;
 	}
-}
+
+	return (
+		<FlatList
+			// rerender FlatList in case of width changes
+			key={`emoji-category-${width}`}
+			// @ts-ignore
+			keyExtractor={item => (item && item.isCustom && item.content) || item}
```

Comment:
```suggestion
			keyExtractor={item => item && item.isCustom ? item.content : item.name}
```

Reply:
This won't work for now as the non-custom emojis are plain strings.